src/client/query.rs

```diff
@@ -425,20 +439,35 @@ impl QueryResultSet {
     pub fn close(&mut self) -> Result<()> {
         match self {
             QueryResultSet::None => Ok(()),
-            QueryResultSet::Some(stream_result) => stream_result.close(),
+            QueryResultSet::Some(stream_result) => {
+                // TODO: async close
+                if stream_result.closed {
+                    Ok(())
+                } else {
+                    Err(CommonErr(
+                        CommonErrCode::Rpc,
+                        "QueryResultSet is not closed".to_owned(),
+                    ))
+                }
+            }
         }
     }
-}
 
-impl Iterator for QueryResultSet {
-    type Item = Result<HashMap<String, Value>>;
+    pub async fn async_close(&mut self) -> Result<()> {
```

Comment:
Why not remove the `close` method, and name this method as `close`?

Reply:
Fixed, only keep the async close now.